### PR TITLE
Re-enable MeshGW tests in acceptance on Kind and update CI binary versions

### PR DESCRIPTION
**Pull request by kschoche**
Changes proposed in this PR:
- Update Helm to `3.9.4` in CI.
- Update Kubernetes versions to latest `1.24.x` for the Kind based acceptance tests.
- Update Kind to `0.15.0`.
- Enable MeshGW tests for Kind.

How I've tested this PR:
CI is passing

How I expect reviewers to test this PR:
👀 

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: acceptance/tests/vault/main_test.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package vault
 
 import (
+	"fmt"
 	"os"
 	"testing"
```
Comment (kschoche) [inline acceptance/tests/vault/main_test.go:7]: I can back this one out, but right now the Vault tests are flaky everywhere, we have a ticket to address them.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1274,13 +1274,13 @@ workflows:
...
 
   nightly-acceptance-tests-consul:
-    triggers:
-      - schedule:
-          cron: "0 0 * * *"
-          filters:
-            branches:
-              only:
-                - main
+    # triggers:
```
Comment (jmurret) [inline .circleci/config.yml:1277]: need to re-enable these before we merge.

**Review by t-eckert — APPROVED** [review]
This looks good 👍🏻

**Review by jmurret — APPROVED** [review]
🙏 :shipit:

**Pull request merged**